src/plugins/azgenerator/CodeModelAzImpl.ts

```diff
@@ -162,6 +177,123 @@ export class CodeModelCliImpl implements CodeModelAz {
             } while (this.SelectNextCommandGroup());
         }
     }
+
+    private setParamAzUniqueNames() {
+        this.paramActionNameReference = new Map<Schema, string>();
+        let nameActionReference: Map<string, ActionParam> = new Map<string, ActionParam>();
+        if (this.SelectFirstCommandGroup()) {
+            do {
+                if (this.SelectFirstCommand()) {
+                    do {
+                        let nameParamReference: Map<string, Parameter> = new Map<string, Parameter>();
+                        if (this.SelectFirstMethod()) {
+                            do {
+                                if (this.SelectFirstMethodParameter()) {
+                                    do {
+                                        let paramName = this.MethodParameter_MapsTo;
+                                        let param = this.MethodParameter;
+                                        let originParam = this.MethodParameter;
+                                        let flattenedNames = param?.['targetProperty']?.['flattenedNames'];
+                                        let mapName: Array<string> = [];
+                                        let flattenedLen = -1;
+                                        let paramFlattenedName = param.language['az'].mapsto;
+                                        let tmpName = paramFlattenedName;
+                                        let names = this.Method_Name.split(' ');
+                                        if (flattenedNames) {
+                                            flattenedLen = flattenedNames.length;
+                                            for(let item of flattenedNames) {
+                                                mapName.push(ToSnakeCase(item));
+                                            }
+                                            if (mapName[0] == 'properties' || mapName[0] == 'parameters') {
+                                                delete mapName[0];
+                                            } else if (names.length > 1 && mapName[flattenedLen - 1].toLowerCase() == names[0].replace(/-/g, '')) {
```

Comment:
>names.length > 1 [](start = 55, length = 16)

from curious, az dataset sql create, the names will be [sql, dataset] ?

Reply:
yeah